pkg/build/controller/strategy/docker_test.go

```diff
@@ -89,13 +89,15 @@ func TestDockerCreateBuildPod(t *testing.T) {
 	// container storage
 	// blobs content cache
 	// global CA injection configmap
-	if len(container.VolumeMounts) != 11 {
-		t.Fatalf("Expected 11 volumes in container, got %d", len(container.VolumeMounts))
+	// node pull secrets
+	if len(container.VolumeMounts) != 12 {
+		t.Fatalf("Expected 12 volumes in container, got %d", len(container.VolumeMounts))
```

Comment:
@ricardomaraschini Need to add the node pull secret to the source strategy (sti) and custom strategy builds, too.

Reply:
Thanks! I forgot those. I think we have a problem when it comes to custom strategy as by mounting these secrets one may easily copy them to the outside world, no? 

Our [documentation](https://docs.openshift.com/container-platform/4.1/builds/build-strategies.html#build-strategy-custom-build_build-strategies) states that: 

"Custom builds run with a very high level of privilege and are not available to users by default. Only users who can be trusted with cluster administration permissions should be granted access to run custom builds."

What do you reckon?

Reply:
good question @ricardomaraschini.  my view would be that it's reasonable to leave this out of custom builds for now and force those users to still provide their own pull secrets.  People doing custom builds already have to be pretty sophisticated.

If we get a lot of pushback about the pain that causes we can revisit the decision, since as you note, someone doing custom builds also already has to be a highly trusted user.

Reply:
@bparees that sounds like a fair compromise to me. We need to doc this in JIRA so we make this clear in the release notes.